cmd/frontend/internal/bg/update_permissions.go

```diff
@@ -49,13 +52,24 @@ func UpdatePermissions(ctx context.Context, logger log.Logger, db database.DB) {
 				return errors.Wrap(err, "creating new permissions")
 			}
 
+			roles := []types.SystemRole{types.SiteAdministratorSystemRole, types.UserSystemRole}
+			excludedRoles := collections.NewSet[rtypes.PermissionNamespace](rbacSchema.ExcludeFromUserRole...)
 			for _, permission := range permissions {
-				// Assign the permission to both SITE_ADMINISTRATOR and USER roles. We do this so that we don't break the
-				// current experience and always assume that everyone has access until a site administrator revokes that
-				// access.
-				// Context: https://sourcegraph.slack.com/archives/C044BUJET7C/p1675292124253779?thread_ts=1675280399.192819&cid=C044BUJET7C
+				// Assign the permission to both SITE_ADMINISTRATOR and USER roles. We do this so
+				// that we don't break the current experience and always assume that everyone has
+				// access until a site administrator revokes that access. Context:
+				// https://sourcegraph.slack.com/archives/C044BUJET7C/p1675292124253779?thread_ts=1675280399.192819&cid=C044BUJET7C
+
+				rolesToAssign := roles
+				if excludedRoles.Has(permission.Namespace) {
+					// The only exception to the above rule (at the moment) is Ownership, because it
+					// is clearly a permission which should be explicitly granted and only
+					// SITE_ADMINISTRATOR has it by default. All exceptions can be added to the
+					// `excludeFromUserRole` attribute of RBAC schema.
+					rolesToAssign = roles[:1]
```

Comment:
Seems a little fragile, what if someone changes the order of roles above, and does not look here?

```suggestion
					rolesToAssign = []types.SystemRole{types.SiteAdministratorSystemRole}
```

Reply:
agree, fixing!